### PR TITLE
Added support for keyboard Pila87 / XM87 Pro

### PR DESCRIPTION
**Pull request by HorrorTroll**
## Description

Added support for keyboard Pila87 / XM87 Pro. It's using CKS32F103RBT6 (a clone STM32F103RBT6), more info is on readme files

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/phage_studio/pila87_mod/rules.mk [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (noroadsleft) [inline keyboards/phage_studio/pila87_mod/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Additionally, any .c, .h, or .inc files you authored need to have GPL v2-compatible license headers.

* [Example GPLv2 Header - QMK Firmware Documentation](https://docs.qmk.fm/#/pr_checklist?id=example-gplv2-header)

File: keyboards/phage_studio/pila87_mod/config.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+#define RGB_MATRIX_KEYPRESSES
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#define DISABLE_RGB_MATRIX_BAND_SAT
+#define DISABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
+#define DISABLE_RGB_MATRIX_BAND_SPIRAL_SAT
+#define DISABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
+#define DISABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#define DISABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#define DISABLE_RGB_MATRIX_SPLASH
+#define DISABLE_RGB_MATRIX_SOLID_SPLASH
```
Comment (drashna) [inline keyboards/phage_studio/pila87_mod/config.h:76]: rgb matrix modes are explicitly enabled now.

File: keyboards/phage_studio/pila87_mod/keymaps/horrortroll/keymap_stuff.h [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+                        return false;
+                }
+            }
+            return false;
+	    }
+    return true;
+}
+
+void rgb_matrix_indicators_user(void) {
+    switch (biton32(layer_state)) {
```
Comment (drashna) [inline keyboards/phage_studio/pila87_mod/keymaps/horrortroll/keymap_stuff.h:186]:
```suggestion
    switch (get_highest_layer(layer_state)) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**